src/App.js

```diff
@@ -1,26 +1,80 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import 'bulma/css/bulma.css';
+import FoodBox from './components/FoodBox'
+import foods from './foods.json';
+import AddFood from './components/AddFood'
+import TodaysFoods from './components/TodaysFoods'
+import FoodForm from './components/FoodForm'
+import searchBar from './components/searchBar'
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+
+  constructor(props){
+    super(props); // read more on this
+    this.state = {
+    showForm: false,  
+    foods: foods,
+    filtered: foods,
+    TodaysFoods: { },
+    };
+
+}
+
+  handleAddfood = (event) => {
+    event.preventDefault() // always do this with submit events
+    let foodName = event.currentTarget.food.value
+    let newfoods = [{title:foodName, completed: false}, ...this.state.foods]
+    this.setState({
+      foods: newfoods,
+    })   
+  }
+
+  handleQuantityChange = (event) => {
+    const { name, value, quantity } = event.target;
+    this.setState({
+      name: name,
+      value: value,
+      quantity: quantity
+    })   
+  }
+
+  filteredFood = (e) => {
+    const searchText = e.currentTarget.value 
```

Comment:
when you call this function in the SearchBar component you are passing the string value as parameter, you are not passing the whole event. So again, there is no need to select currentTarge.value, just the e variable will get you the input value.